docs/source/posts/2023/2023-06-12-week-2-joaodellagli.rst

```diff
@@ -0,0 +1,86 @@
+   :author: João Victor Dell Agli Floriano
+   :tags: google
+   :category: gsoc
+
+
+Hello everybody, welcome to the week 2 of this project! I must admit I thought this would be simpler than it is currently being, but I forgot that when it comes to dealing with computer graphics' applications, things never are. Below, some updates on what I have been up to for this past week. 
+
+This Last Week's Effort
+-----------------------
+
+Last week, I was facing some issues with a VTK feature essential so I could move forward with my project: Framebuffer Objects. 
+As described in my `last blogpost <https://blogs.python-gsoc.org/en/joaodellaglis-blog/the-fbo-saga-week-1/>`_, for some reason the 2D allocation methods for it weren't working.
+In a meeting with my mentors, while we were discussing and searching through VTK's FramebufferObject and TextureObject documentation, and the code itself for the problem, 
+one TextureObject method caught my attention: `vtkTextureObject.SetContext() <https://vtk.org/doc/nightly/html/classvtkTextureObject.html#a0988fa2a30b640c93392c2188030537e>`_.
+
+Where the Problem Was
+---------------------
+My last week's code was:
+
+::
+
+   color_texture = vtk.vtkTextureObject() # color texture declaration
+   color_texture.Bind() # binding of the texture for operations
+
+   color_texture.SetDataType(vtk.VTK_UNSIGNED_CHAR) # setting the datatype for unsigned char
+   color_texture.SetInternalFormat(vtk.VTK_RGBA) # setting the format as RGBA
+   color_texture.SetFormat(vtk.VTK_RGBA)
+   color_texture.SetMinificationFilter(0) # setting the minfilter as linear
+   color_texture.SetMagnificationFilter(0) # setting the magfilter as linear
+
+   color_texture.Allocate2D(width, height, 4, vtk.VTK_UNSIGNED_CHAR) # here is where the code stops
+
+But it turns out that to allocate the FBO's textures, of type vtkTextureObject, you need to also set the context where the texture object
+will be present, so it lacked a line, that should be added after ``Bind()``:
+
+::
+
+   color_texture = vtk.vtkTextureObject() 
+   color_texture.Bind() 
+
+   color_texture.SetContext(manager.window) # set the context where the texture object will be present
+
+   color_texture.SetDataType(vtk.VTK_UNSIGNED_CHAR) 
+   color_texture.SetInternalFormat(vtk.VTK_RGB) 
+   color_texture.SetFormat(vtk.VTK_RGB)
+   color_texture.SetMinificationFilter(0) 
+   color_texture.SetMagnificationFilter(0) 
+
+The code worked fine. But as my last blogpost showed, Allocate3D() method worked just fine without a (visible) problem, why is that? 
+Well, in fact, it **didn't work**. If we check the code for the Allocate2D and Allocate3D, one difference can be spotted:
+
+
+
+.. image:: https://raw.githubusercontent.com/JoaoDell/gsoc_assets/main/images/allocate-2d-3d.png
+   :align: center
+   :alt: Image comparing Allocate2D and Allocate3D methods
+
+
+
+While in Allocate2D there is an ``assert(this->Context);``, in Allocate3D the assertion is translated into:
```

Comment:
``Allocate2D``   
``Allocate3D``

mark them everywhere like you did for  ``assert(this->Context);``